tests/user/test_get_user_by_id.py

```diff
@@ -0,0 +1,230 @@
+import pytest
+from allure import feature, description, link, step, title
+from hamcrest import assert_that, is_, contains_string, is_not, empty
+
+from framework.asserts.common import assert_status_code, assert_content_type, assert_response_message
+from framework.asserts.user_asserts import assert_all_user_data_matches
+from framework.endpoints.authenticate_api import AuthenticateAPI
+from framework.endpoints.users_api import UsersAPI
+from framework.tools.generators import generate_user
+from framework.tools.jwt_token import generate_jwt_token
+
+
+@feature("Getting user info by ID")
+@link(
+    url="https://github.com/Sunagatov/Online-Store/wiki/",
+    name="(!) WAIT LINK. Description of the tested functionality",
+)
+class TestGetUserById:
+    @title("Getting User's Own Information by ID")
+    @description(
+        "GIVEN the user is logged in, "
+        "WHEN the user sends a request to get their own information by ID, "
+        "THEN the response code is 200 and the response body contains the current user's data."
+    )
+    def test_get_user_self_info_with_valid_id(self, create_authorized_user):
+        user, token = [create_authorized_user["user"], create_authorized_user["token"]]
+
+        with step("Getting user info by ID via API"):
+            getting_user_response = UsersAPI().get_user_by_id(token=token, user_id=user["id"])
+
+        with step("Checking the response code"):
+            assert_status_code(getting_user_response, 200)
+
+        with step("Checking the response body"):
+            user_data = getting_user_response.json()
+            assert_all_user_data_matches(user_data, user)
+
+    @title("Getting Another User Info by ID")
+    @description(
+        "GIVEN the user is logged in, "
+        "WHEN the user sends a request to get information about another user by ID, "
+        "THEN the response code is 200 and the response body contains the searched user's data"
+    )
+    def test_get_another_user_info_with_valid_id(self, postgres, create_authorized_user):
+        with step("Creating another user"):
+            another_user = generate_user()
+            postgres.create_user(another_user)
+
+        with step("Getting another user info by ID via API"):
+            token = create_authorized_user["token"]
+            getting_user_response = UsersAPI().get_user_by_id(token=token, user_id=another_user["id"])
+
+        with step("Checking the response code"):
+            assert_status_code(getting_user_response, 200)
+
+        with step("Checking the response body"):
+            user_data = getting_user_response.json()
+            assert_all_user_data_matches(user_data, another_user)
+
+    @pytest.mark.skip(reason="NEED TO CLARIFY: Bug in the API => wrong response type (text instead of JSON)")
```

Comment:
Bug report? Task?

Reply:
This bug is already in the temporary Excel file for bugs until Jira readiness.